### ISSUE TITLE
[wgsl-in] Parse error for subtraction in array index

### ISSUE THREAD

**Issue by fintelia** [gfx-rs/wgpu#4435]
This shader:
```rust
var<workgroup> a: array<u32, 64>;

fn foo() {
    let x = 1;
    let y = a[x-1];
}
```
Produces this error:

```
error: expected ']', found '-1'
  ┌─ a.wgsl:6:16
  │
6 │     let y = a[x-1];
  │                ^^ expected ']'

Could not parse WGSL
```

**Comment by jimblandy** [comment]
The issue here is that Naga treats `-1` as a token, not an application of a unary `-` operator to the literal `1`, so the expression is sort of like `a[x 7]`. This used to be standard behavior, but the WGSL specification changed.

**Comment by jimblandy** [comment]
This will be fixed by #4870.
